docs/modules/maintain-cluster/pages/enterprise-rest-api.adoc

```diff
@@ -0,0 +1,460 @@
+
+* **#Enable Security:#**
+You need to enable the security to use REST API.
+
+* **#Configure Rest Realm:#**
+You need to introduce a REST realm as the authentication mechanism. This realm will handle user authentication using a simple method based on username and password.
+
+* **#Setup Users and Roles:#**
+You should identify users and their roles within the REST realm.
+
+* **#Configure AccessControlService:#**
+To configure `AccessControlService`, you should provide a factory class that creates an instance of the service. `AccessControlService` delegates authentication to the configured security realm. You can either create a custom factory class, or use the default implementation. For the default implementation, you can use `DefaultAccessControlServiceFactory`, to produce `DefaultAccessControlService`. Authorization decisions in this service are based on three predefined roles: `reader`, who can make requests via GET endpoints only; `writer`, who can make requests including GET, POST, and DELETE; and `admin`, who can make all requests and access admin-specific endpoints.
+
+Following declarative example includes all the steps:
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+    <security enabled="true">
+        <realms>
+            <realm name="restRealm">
+                <authentication>
+                    <simple>
+                        <user username="restuser" password="restsecret">
+                            <role>admin</role>
+                        </user>
+                    </simple>
+                </authentication>
+                <access-control-service>
+                    <factory-class-name>com.hazelcast.internal.rest.access.DefaultAccessControlServiceFactory</factory-class-name>
+                </access-control-service>
+            </realm>
+        </realms>
+    </security>
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  security:
+    enabled: true
+    realms:
+      - name: restRealm
+        authentication:
+          simple:
+            users:
+              - username: 'restuser'
+                password: 'restpassword'
+                roles:
+                  - admin
+        access-control-service:
+          factory-class-name: com.hazelcast.internal.rest.access.DefaultAccessControlServiceFactory
+----
+====
+
+=== Authentication Process
+
+To authenticate to the REST API, the REST client must obtain a token. See the <<obtaining-a-token, Obtaining a Token section>> for details. The token request requires a username and password and the role(s) assigned to the user are included in the token as claims. You can use the token until it expires. You use the token at the `Authorization` header for each subsequent REST API call so that the call is authorized appropriately. If a valid token is not included in the header, the API will respond with a `401 Unauthorized` error.
```

Comment:
Couldn't understand the "and the role(s) assigned to the user are included in the token as claims". Does this mean the response (token) automatically knows the role for the user having the username/password configured within the security realm?

Reply:
When generating a token for a user, we include their assigned roles as a claims. When user sends a request with the token, we parse it to get their assigned roles.